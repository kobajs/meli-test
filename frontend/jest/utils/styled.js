import { shallow, mount } from "enzyme";
import ThemeProvider from "../../src/theme/ThemeProvider";

function wrapWithTheme(fn, children, options) {
  const wrapper = fn(withTheme(children), options);

  return wrapper;
}

export function withTheme(children) {
  return <ThemeProvider>{children}</ThemeProvider>;
}

export function shallowWithTheme() {
  return wrapWithTheme(shallow, ...arguments);
}

export function mountWithTheme() {
  return wrapWithTheme(mount, ...arguments);
}
