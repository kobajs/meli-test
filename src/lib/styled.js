import get from 'lodash/get'

export function theme(path) {
  return props => get(props, `theme.${path}`);
}

export function breakpoints(breakpoint) {
  return props => `@media (min-width: ${props.theme.breakpoints[breakpoint]}px)`;
}