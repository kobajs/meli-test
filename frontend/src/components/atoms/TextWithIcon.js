import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import { theme } from "../../lib/styled";

const Wrapper = styled.div`
  display: flex;
  align-items: center;
`

const Icon = styled.img`
  height: 20;
  width: 20;
  margin: 0 ${theme("spacing.unit")}px;
`;

function TextWithIcon({ className, children, icon, alt }) {
  return (
    <Wrapper className={className}>
      <span>{children}</span>
      <Icon src={icon} alt={alt} />
    </Wrapper>
  );
}

TextWithIcon.propTypes = {
  children: PropTypes.string.isRequired,
  icon: PropTypes.string.isRequired,
  alt: PropTypes.string.isRequired
};

export default styled(TextWithIcon)``;
