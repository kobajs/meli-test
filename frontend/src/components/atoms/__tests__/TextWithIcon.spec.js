import React from "react";
import renderer from "react-test-renderer";

import TextWithIcon from "../TextWithIcon";
import { withTheme } from "../../../../jest/utils/styled";

it("renders correctly", () => {
  const ThemedTextWithIcon = withTheme(
    <TextWithIcon icon="/static/logo.png" alt="logo">
      Test TextWithIcon
    </TextWithIcon>
  );
  const tree = renderer.create(ThemedTextWithIcon).toJSON();
  expect(tree).toMatchSnapshot();
});
