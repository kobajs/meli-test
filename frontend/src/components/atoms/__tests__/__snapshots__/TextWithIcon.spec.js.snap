// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="TextWithIcon-v7x5c0-2 gxLptL TextWithIcon__Wrapper-v7x5c0-0 ePVtQK"
>
  <span>
    Test TextWithIcon
  </span>
  <img
    alt="logo"
    className="TextWithIcon__Icon-v7x5c0-1 bkxtfY"
    src="/static/logo.png"
  />
</div>
`;
